code/script.js

```diff
@@ -221,123 +268,244 @@ const CHARACTERS = [
+        attribute
+    } 
+    category = 'hairColor'
+  } else if (category === 'eye color') { 
+    question = {
+      category: 'eyeColor',
+      attribute 
     }
-  } else if (category === 'eye color') {
-    // Set this up your self
+    category = 'eyeColor'
   } else if (category === 'accessories') {
-    currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      if ('glasses' === attribute) {
+        question = {
+          category: 'glasses',
+          attribute: true
+        }
+      }
+      else if ('hat' === attribute) {
+        question = {
+          category: 'hat',
+          attribute: true
+        }
+      } else if ('earrings' === attribute) {
+        question = {
+          category: 'earrings',
+          attribute: true
+      }
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+      if ('smoker' === attribute) {
+      question = {
+        category: 'smoker',
+        attribute: true
+      } 
+    } else if ('beard' === attribute) {
+      question = {
+        category: 'beard',
+        attribute: true
+      } 
+    }
   }
-}
+    console.log(question);
+    currentQuestion = question
 
-// This function should be invoked when you click on 'Find Out'.
-const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-}
+const keep = secret[question.category] === question.attribute 
+//console.log(secret[question.category], secret)
+console.log('You are one step closer to finding out who the secret person is!', keep) 
 
-// It'll filter the characters array and redraw the game board.
-const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+    // Om den hemliga personen är samma som nuv. kategori och är densamma som nuv. valt värde filtrera Sant, else filtrera Falskt. 
+    if (keep === true) {
+      console.log('You matched')
+      filterCharacters(keep, question.category, question.attribute)
+    } else {
+      console.log('You guess is not right this time, please try again!') 
+      filterCharacters(keep, question.category, question.attribute) //pass by value
+    }
+  }
+
+//question.category is now group, and question.attribute is now attribute
+const filterCharacters = (keep, group, attribute) => {
+  console.log(keep, group, attribute);
+  if (keep != true) { 
+    let charactersFiltered = charactersInPlay.filter(name => {
+      if (name[group] === attribute) { 
+        return false
+      } else {
+        return true
+      }
+    });
+    charactersInPlay = charactersFiltered
+  }
```

Comment:
Yes this is a bug that needs to be fixed to meet the requirements 🤓

Reply:
I started out with a filtering that took away characters also when the answer is true, but decided to change this so when the user chooses the right value no filtering will happen. It should only filter on false. 

I don't consider it a bug, but as part of the game. If a user would choose eyeglasses and all characters with glasses were filtered away on that click, except for the one that was the secret person the game would be too easy. That is how I understood and programmed the game. If you want me to change this I can change it, but I think that the game is too easy then.

Reply:
Hello, I did not mean to sound arrogant. I just understood the rules of the game in this way and wrote it so that it would not filter any characters if a guess was true. Is this still considered a bug?

Reply:
I changed the filter characters function to meet the requirements.